week2/react-exercises/dog/src/App.js

```diff
@@ -0,0 +1,57 @@
+import React, {useState, useEffect} from 'react'
+import './App.css'
+
+const Button = ({getDogPhoto}) => {
+  return (
+    <button onClick={getDogPhoto}>Get a dog!</button>
+  )
+}
+
+const DogPhoto = ({dogPhoto}) => {
+  return (
+    <>
+      <img className="c-image" src={dogPhoto} alt="Dog"></img>
+    </>
+  )
+}
+
+const API_URL ='https://dog.ceo/api/breeds/image/random';
+
+const DogGallery = () => {
+
+  const [dogPhotos, setDogPhotos] = useState([]);
+
+  const getDogPhoto = async () => {
+    const response = await fetch(API_URL);
+    const data = await response.json();
+    const item = data.message;
+    setDogPhotos([...dogPhotos, item]);
```

Comment:
Suggestion in order not to have a dependency on the current dogPhotos
```suggestion
    setDogPhotos((prevDogPhotos) => ([...prevDogPhotos, item]));
```

Reply:
Great point!